RecursiveFileSearch

Stalfy opened the pull request:
Modified file search procedure to be more flexible.

Stalfy committed:
Extracted filesearch from interpreter folder.

Stalfy committed:
Changed to recursive file search.

Stalfy committed:
Removed console.log. Logger required.

Stalfy committed:
Modified gitignore to ignore htmlOutput folder.

Stalfy committed:
Added markdown files to acceptance test to test recursion.

Grasseh reviewed (commented):
Sweet work. Assuming the Conflict fixed will be as good~

Grasseh commented on .pferc:
```diff
@@ -1,4 +1,4 @@
 {
-    "outputPath" : "/home/stalf/Projects/PFE_H_2019/demo/js/markdown/output",
-    "basePath" : "/home/stalf/Projects/PFE_H_2019/demo/js/markdown"
+    "outputPath" : "/home/stalf/Projects/PFE_H_2019/demo/js/htmlOutput",
```
This file should probably be gitignored.

Although we gotta be careful to NOT ignore the one in tests!

Grasseh commented on src/filesearch/MarkdownFileLocator.js:
```diff
@@ -0,0 +1,39 @@
+
+    _scanPath(basePath) {
+        let ext = path.extname(basePath);
+        if('' !== ext && DESIRED_EXT !== ext) {
```
Wouldn't it be better to check if the ext is not a different extension than empty or .md somewhere else and give out a clean error?

Kinda odd to give an empty array if I fave myfile.mdd (or if someone uses odd extensions for their markdowns although they still have fine syntax)

Lolskate replied:
https://superuser.com/questions/249436/file-extension-for-markdown-files

We should allow the ones present in this list, it would be less confusing for people who use them.

Stalfy replied:
Updated to match some common extensions.

Grasseh reviewed (commented):
Two Nitpicks

Stalfy committed:
Ignored local taffeerc.

Stalfy committed:
Removed old .pferc config

Stalfy committed:
Fixed acceptance tests.

Stalfy committed:
Uniformized test structure.

Stalfy committed:
Removed unused GenericParser (YAGNI).

KB07 reviewed (approved):
À part qu'il faudrait aussi des input/ouput test pour demo/php/... 
All seems good to me.

Stalfy committed:
Modified output integration test to use real classes.

Lolskate reviewed (dismissed):
See comment under Steve's

Stalfy committed:
Renamed non-explicit package name testsuite to model.

Stalfy committed:
Removed index.js from main src file.
Renamed interpreter package to parser to reflect content.
Renamed filesearch package to locator to reflect content.
Restructured affected tests and artifacts.

Stalfy committed:
Removed unused index.js files.

Stalfy committed:
Updated accepted extensions for markdown.

Stalfy committed:
Modified run.js to replacy any extension properly.

Stalfy committed:
Modified run.js to use getMarkdown() method from test suite result.

Stalfy committed:
Added cssFiles and templates to acceptation test.

Stalfy committed:
Remove stray console.log.

Grasseh commented on src/locator/MarkdownFileLocator.js:
```diff
@@ -0,0 +1,35 @@
+
+class MarkdownFileLocator {
+    locateFiles(basePath) {
+        if (false === fs.existsSync(basePath)) {
```
replacing false === by ```!fs.existsSync(basePath)``` would be better?

Stalfy replied:
Added to changes.

Grasseh commented on test/acceptation/app.js:
```diff
@@ -6,17 +6,18 @@ const path = require('path');
     it('Should work from A to Z', function() {
-        this.timeout(4000);
-        let configPath = path.join(__dirname, '..', 'integration', 'artifacts', 'application', '.pferc');
+        this.timeout(5000);
```
At this point make this test 50000 hahahaha

Stalfy replied:
Just to be sure travis would not explode.

Grasseh replied:
Yeah that's my point hahaha Travis will not explode for a while if we make it 50000

Grasseh commented on test/acceptation/artifacts/test1.md:
```diff
@@ -1,5 +1,5 @@
-[](i:PhpInvoker)
-[](m:../testModule.php)
+[](i: PhpInvoker)
```
This works?

Stalfy replied:
Damn right it does.

Grasseh commented on test/integration/invoker/NodeInvoker.js:
```diff
@@ -9,7 +9,7 @@ describe('NodeInvoker Integration', function() {
             let testName = 'myTestFn';
             let params = [1, 2];
-            let projectName = path.join(process.cwd(), 'test', 'integration', 'my artifacts', 'testModule');
+            let projectName = path.join(process.cwd(), 'test', 'integration', 'artifacts', 'invoker', 'testModule');
```
I'd rather still have an artifact name with spaces somewhere just to test that spaces are handles correctly

Stalfy replied:
Added tests to test this particular case.

Grasseh reviewed (changes requested):
Few nitpicks here and there

KB07 reviewed (approved):
No nitpick for me

Stalfy committed:
Added tests to verify the support of spaced directories. Fixed some nitpicks.
